src/block/get.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-env serviceworker, browser */
+/* global Response */
+
+import { base58btc } from 'multiformats/bases/base58'
+
+import { BlockNotFoundError, BaseNotFoundError } from '../errors.js'
+
+/**
+ * @typedef {import('../env').Env} Env
+ */
+
+/**
+ * Handle block get request
+ *
+ * @param {Request} request
+ * @param {Env} env
+ * @param {import('./index').Ctx} ctx
+ */
+export async function blockGet (request, env, ctx) {
+  // Get cached block if exists
+  const cache = caches.default
+  let res = await cache.match(request)
+  if (res) {
+    return res
+  }
+
+  const multihash = request.params.multihash
+
```

Comment:
I think it would read easier to if base transcoding was factored into separate function e.g.

```js
const key = toBase58btc(multihash)
const r2Object = await env.BLOCKSTORE.get(key)
// ....
```